config/src/main/java/com/typesafe/config/impl/Tokens.java

```diff
@@ -180,8 +198,8 @@ public int hashCode() {
     static private class Comment extends Token {
         final private String text;
 
-        Comment(ConfigOrigin origin, String text) {
-            super(TokenType.COMMENT, origin);
+        Comment(ConfigOrigin origin, String text, boolean doubleSlash) {
+            super(TokenType.COMMENT, origin, (doubleSlash? "//" : "#") + text);
```

Comment:
this would be a little nicer I think if we do it via class hierarchy; `abstract class Comment extends Token`, `DoubleSlashComment extends Comment`, `HashComment extends Comment`, or something. It saves a tiny bit of ram, but the more important reason is that booleans don't read very well. Later on when you use this boolean in the test suite it's hard to know what it means.  long version at http://blog.ometer.com/2011/01/20/boolean-parameters-are-wrong/